ml-proto/spec/memory.ml

```diff
@@ -79,9 +80,13 @@ let grow mem n =
   Array1.blit (Array1.sub !mem 0 host_old_size) (Array1.sub after 0 host_old_size);
   mem := after
 
-let rec loadn mem n a =
+let effective_address a o =
+  if (Int64.sub Int64.max_int a) < o then raise Bounds;
```

Comment:
This is doing a signed comparison. The design text describes the effective address as an unsigned computation, so it seems like this should be checking for unsigned overflow instead.